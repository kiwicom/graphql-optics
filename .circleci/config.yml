version: 2
jobs:
  build:
    docker:
      # specify the version you desire here
      - image: circleci/node:9.11
    working_directory: ~/repo
    steps:
      - checkout
      - restore_cache:
          keys:
          - v1-dependencies-{{ checksum "package.json" }}
          - v1-dependencies-
      - run: yarn install
      - run: cd packages/tracer && yarn install
      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}
      - run: yarn test
