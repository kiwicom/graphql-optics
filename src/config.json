{
  "elastic": {
    "host": "localhost:9200",
    "log": "trace"
  }
}
