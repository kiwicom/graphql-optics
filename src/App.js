import React from 'react'
import AggregationList from './AggregationList'
import server from './server'
import config from './config.json'

class App extends React.Component {
  loadAggregations() {
    server
      .search({
        'index': config.elastic_index,
        'body': {
          'size' : 0,
          'aggs' : {
            'groups' : {
              'terms' : {
                'field': 'group'
              }
            }
          }
        }
      })
      .then((data) => {
        this.state = {
          ...this.state,
          aggregations: data.aggregations.groups.buckets
        }
      })
  }

  constructor(props) {
    super(props)
    this.state = {
      aggregations: []
    }
    this.loadAggregations()
  }

  render() {
    return (<div>
      <h1>graphql-optics</h1>
      <div>
        <h2>Aggregations</h2>
        <AggregationList aggregations={this.state.aggregations} />
      </div>
    </div>)
  }
}

export default App
