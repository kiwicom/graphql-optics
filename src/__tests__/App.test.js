import React from 'react'
import { renderWithRouter } from './helpers'
import App from '../App'

import config from '../config.json'

jest.mock('../server', () => {
  const data = [
    {
      key: 'allLocations',
      doc_count: 442,
    },
    {
      key: 'allFlights',
      doc_count: 3342,
    },
    {
      key: 'allWhatever',
      doc_count: 2,
    },
  ]

  const _search_spy = jest.fn()

  const aggsResponse = {
    aggregations: {
      rootQueries: {
        buckets: data,
      },
    },
  }
  const queriesResponse = {
    hits: {
      hits: [{ _id: 'uuid', _source: { graphql: 'allLocations...' } }],
    },
  }

  return {
    search: query => {
      _search_spy(query)
      return {
        then: fn => fn(query.body.aggs ? aggsResponse : queriesResponse),
      }
    },
    _search_spy,
  }
})

afterEach(() => {
  jest.resetAllMocks()
})

describe('App', () => {
  it('renders correctly (after data is returned from server)', () => {
    const tree = renderWithRouter({ children: <App /> })
    expect(tree).toMatchSnapshot()
  })

  it('requests aggregation list using the correct query', () => {
    const tree = renderWithRouter({ children: <App /> })
    const server = require.requireMock('../server')
    expect(server._search_spy).toHaveBeenCalledWith({
      index: config.elastic_index,
      body: {
        size: 0,
        aggs: {
          rootQueries: {
            terms: {
              field: 'rootQuery',
            },
          },
        },
      },
    })
  })

  it('requests query list using the rootQuery route param', () => {
    const tree = renderWithRouter({
      children: <App />,
      location: '/aggregations/allLocations',
    })
    const server = require.requireMock('../server')
    expect(server._search_spy.mock.calls[1][0]).toEqual({
      index: config.elastic_index,
      body: {
        query: {
          match: {
            rootQuery: 'allLocations',
          },
        },
      },
    })
  })
})
