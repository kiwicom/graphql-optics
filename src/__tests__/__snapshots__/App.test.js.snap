// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly (after data is returned from server) 1`] = `
<div>
  <h1>
    graphql-optics
  </h1>
  <div>
    <h2>
      Aggregations
    </h2>
    <ul>
      <li>
        allLocations
         - 
        442
      </li>
      <li>
        allFlights
         - 
        3342
      </li>
      <li>
        allWhatever
         - 
        2
      </li>
    </ul>
  </div>
</div>
`;
